assets/js/data/cart/reducers.js

```diff
@@ -11,6 +11,15 @@ import { ACTION_TYPES as types } from './action-types';
  */
 const cartItemsReducer = ( state = [], action ) => {
 	switch ( action.type ) {
+		case types.RECEIVE_CART_ITEM:
+			// Replace specified cart element with the new data from server.
+			return state.map( ( cartItem ) => {
+				if ( cartItem.key === action.cartItem.key ) {
+					return action.cartItem;
+				}
+				return cartItem;
+			} );
```

Comment:
This will need to change because while it updates the cart item in the `cartData.cartItems` state, the totals are not updated. We'll need to resolve what endpoint you use (and thus how the reducer functions for this) in #1840

Reply:
Agreed, I'm assuming you mean this will change in a future PR (not blocking this PR). Perhaps these piecemeal `RECIEVE_*` actions will just get removed, and we'll only need `RECIEVE_CART`.

Reply:
> Agreed, I'm assuming you mean this will change in a future PR (not blocking this PR).